ggml/src/ggml-common.h

```diff
@@ -418,6 +418,60 @@ typedef struct {
 } block_iq4_xs;
 static_assert(sizeof(block_iq4_xs) == sizeof(ggml_half) + sizeof(uint16_t) + QK_K/64 + QK_K/2, "wrong iq4_xs block size/padding");
 
+#if defined(GGML_COMMON_DECL_METAL_KARGS)
+typedef struct {
```

Comment:
I think it would be preferable to create a new header for the Metal backend than adding this here, this isn't common to multiple backends. Generally we should aim to keep coupling between the common ggml files and the backends to a minimum to make the code easier to maintain. I plan to move each backend to a different directory as part of the changes to the build system that I am working on.

Reply:
Will do 👍

Reply:
PTAL at [ab6a3b7](https://github.com/ggerganov/llama.cpp/pull/10238/commits/ab6a3b7c36c1e90ea0fbce79e89daa103f7be562) if it's OK.